tutor/src/helpers/task-plan.coffee

```diff
@@ -31,4 +32,16 @@ module.exports = {
   earliestDueDate: (plan) ->
     dates = _.map(plan?.tasking_plans, 'due_at')
     _.first(dates.sort()) or ''
+
+
+  apiEndpointOptions: (id, courseId) ->
+    task = TaskPlanStore.get(id)
+    options = if TaskPlanStore.isNew(id)
+      url: "courses/#{courseId}"
+      method: 'POST'
+    else
+      url: "plans/#{id}"
```

Comment:
the query can be sent as

```coffee
params: 
  ecosystem_id: task.ecosystem_id
```

as axios will turn it into a query string: https://github.com/mzabriskie/axios#example